compareMemoryLimit fails in TestKillOneDataNode

E2E test TestKillOneDataNode fails in [CI](https://devops-ci.elastic.co/job/cloud-on-k8s-e2e-tests/11/console):

```go
08:28:33      --- FAIL: TestKillOneDataNode/Elasticsearch_nodes_topology_should_be_the_expected_one (0.04s)
08:28:33          require.go:126: 
08:28:33              	Error Trace:	checks_es.go:149
08:28:33              	Error:      	Should be empty, but was [{ 0xc0003cc8f0 1 {{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {[] [] [{elasticsearch  [] []  [] [] [] {map[cpu:{{2 0} {<nil>} 2 DecimalSI} memory:{{2 9} {<nil>} 2G DecimalSI}] map[]} [] [] nil nil nil    nil false false false}]  <nil> <nil>  map[]   <nil>  false false false <nil> nil []   nil  [] []  <nil> nil [] <nil> <nil>}} []} { 0xc0003cc8f8 2 {{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {[] [] [{elasticsearch  [] []  [] [] [] {map[memory:{{2 9} {<nil>} 2G DecimalSI} cpu:{{2 0} {<nil>} 2 DecimalSI}] map[]} [] [] nil nil nil    nil false false false}]  <nil> <nil>  map[]   <nil>  false false false <nil> nil []   nil  [] []  <nil> nil [] <nil> <nil>}} []} { 0xc0003cc8f8 2 {{      0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] nil [] } {[] [] [{elasticsearch  [] []  [] [] [] {map[memory:{{2 9} {<nil>} 2G DecimalSI} cpu:{{2 0} {<nil>} 2 DecimalSI}] map[]} [] [] nil nil nil    nil false false false}]  <nil> <nil>  map[]   <nil>  false false false <nil> nil []   nil  [] []  <nil> nil [] <nil> <nil>}} []}]
```

https://github.com/elastic/cloud-on-k8s/blob/0a760c98188a96e7cc69d77eb506f1f02ca746b0/operators/test/e2e/stack/checks_es.go#L148-L149

The expected topology does not match all nodes because 

https://github.com/elastic/cloud-on-k8s/blob/0a760c98188a96e7cc69d77eb506f1f02ca746b0/operators/test/e2e/stack/checks_es.go#L139

is always `false`.

The `topoElem` defines a memory limit of 2000000000 while `cgroupMemoryLimitsInBytes` returns by ES is 1999998976.

```bash
> k -n e2e get po test-failure-kill-one-data-node-es-h78tnt68hk -o json | jq '.spec.containers[].resources.requests'       
{
  "cpu": "2",
  "memory": "2G"
}
```

```bash
> k exec -n e2e -ti test-failure-kill-one-data-node-es-h78tnt68hk bash
[root@test-failure-kill-one-data-node-es-h78tnt68hk elasticsearch]# cat /sys/fs/cgroup/memory/memory.limit_in_bytes
1999998976
```

--- a/operators/test/e2e/stack/builder.go
+++ b/operators/test/e2e/stack/builder.go
@@ -18,7 +18,7 @@ import (
 
 var DefaultResources = corev1.ResourceRequirements{
 	Limits: map[corev1.ResourceName]resource.Quantity{
-		corev1.ResourceMemory: resource.MustParse("2G"),
+		corev1.ResourceMemory: resource.MustParse("2Gi"),
 		corev1.ResourceCPU:    resource.MustParse("2"),
 	},
 }

